isis/src/base/apps/phocube/main.cpp

```diff
@@ -333,42 +319,38 @@ void IsisMain() {
 
   UpdateBandKey("Width", bb, nvals, "1.0");
 
-
-
   p.EndProcess();
 }
 
-
-//  This propagates the input plane to the output plane, then passes it off to
-//  the general routine
-void phocubeDN(Buffer &in, Buffer &out) {
-  for (int i = 0 ; i < in.size() ; i++) {
-    out[i] = in[i];
-  }
-  phocube(out);
-}
-
-
 //  Computes all the geometric properties for the output buffer.  Certain
 //  knowledge of the buffers size is assumed below, so ensure the buffer
 //  is still of the expected size.
-void phocube(Buffer &out) {
-
-
-  // If the DN option is selected, it is already added by the phocubeDN
-  // function.  We must compute the offset to start at the second band.
-  int skipDN = (dn) ? 64 * 64   :  0;
-
+void phocube(Buffer &in, Buffer &out) {
   for (int i = 0; i < 64; i++) {
     for (int j = 0; j < 64; j++) {
 
       MosData mosd, *p_mosd(0);  // For special mosaic angles
 
-      int index = i * 64 + j + skipDN;
+      int index = i * 64 + j;
+      int inIndex = index;
+
+      if (dn) {
+        out[index] = in[index];
+        index += 64 * 64;
+      }
+
+      // If dn is true, make sure to not overwrite the original pixels with NULL for the dn band
+      if (!specialPixels && IsSpecial(in[inIndex])) {
+        for (int band = (dn) ? 1 : 0; band < nbands; band++) {
+          out[index] = Isis::NULL8;
```

Comment:
Why NULL8 and not just NULL?

Reply:
I was following the behavior of setting pixels off the body to NULL8 on line 527. Should this be set to NULL and we leave the current behavior of the off-body pixels? Or should they both be NULL?

Reply:
I mentioned this in the stand-up but I want it documented here for future folks. These should all be changed to Isis::Null because they are not dependent upon the bit-type of the image. In general, we use the specific bit-type special pixel values when the logic depend on the bit-type of the image and the standard special pixel values when the logic is valid for all bit-types.